compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -3172,11 +3166,11 @@ public BLangNode transform(StartActionNode startActionNode) {
         BLangNode expression = createActionOrExpression(startActionNode.expression());
 
         BLangInvocation invocation;
-        if (!(expression instanceof BLangWorkerSend)) {
+        if (!(expression instanceof BLangWorkerAsyncSendExpr)) {
```

Comment:
I know the code was there already, but checking `instanceof` is not a good practice. Please check the NodeKind instead. like below expression.getKind() == NodeKind.WORKER_SEND.. etc

Reply:
Updated 40cc512e09d1ce6989618c1c7392ab833ab772b5